crates/uv-resolver/src/pubgrub/report.rs

```diff
@@ -539,6 +538,14 @@ impl PubGrubReportFormatter<'_> {
                         incomplete_packages,
                         output_hints,
                     );
+
+                    // Check for no versions on the local packages themself, which may
```

Comment:
Ugh, this logic is too weak and gets tricked into false positives, I think from the root itself appearing as a normal package down the chain even on other kind of failures.
I think I need to enhance this heuristic in order to detect the cycle.